main.py

```diff
@@ -34,21 +33,10 @@ async def new_message_listener(client, event):
     lemmatizer = Lemmatizer()
     lemmas = lemmatizer.lemmatize(event.text + ' ' + event.text.replace('-', ''))
 
-    matched_keywords = set()
-
-    # Calculate intersections and apply rules based on the new groups structure
-    for group in YamlProcessor.groups_data:
-        intersection_keywords = lemmas.intersection(group.keywords)
-        intersection_include = lemmas.intersection(group.include_keywords)
-        intersection_exclude = lemmas.intersection(group.exclude_keywords)
-
-        if intersection_keywords:
-            if (group.include_keywords and intersection_include) or (
-                    group.exclude_keywords and not intersection_exclude):
-                matched_keywords.update(intersection_keywords)
+    matched_keywords = find_intersections(lemmas)
 
     post_message_to_db_archive(event.text, (' ').join(lemmas), f"https://t.me/{event.chat.username}/{event.id}",
-                               bool(matched_keywords), EnvProcessor.message_collecting_is_on)
+                               bool(matched_keywords), EnvProcessor.message_collecting_is_on())

```

Comment:
Just curious, why to keep `message_collecting_is_on` as a method?

Reply:
That's because it's not perfect to keep using @property for class methods since it's not obvious that when the property is called python implicitly retrieves its value, without Property type wrapper (and IDE will complain about that as well). After all, property is much more suitable for instance, not class methods. That's why I've decided to get rid of property decorator

Reply:
I mean this:
![image](https://github.com/catebi/FreeganBot/assets/39305933/e1adc1ea-6a66-4314-8a13-f600f38db0cc)

Reply:
> Just curious, why to keep `message_collecting_is_on` as a method? same for some other similar parameters, for which there are methods with just `return` in the body

upd: okay, it's instead of @property 
but maybe would be better to return how it used to be, as constants imported from .env? Or even to read them from the db, so we could update settings without touching the code

Reply:
They are still imported from .env just as it was with "os.getenv", it has just a generic way to do that.
Regarding to DB, our env variables do not have universal values to sore them in db since each user has to insert their personal data, chat urls and so on. Even message collecting can be an option on local development. So I don't think that we should store them not in .env but in DB

Reply:
They can be stored and read either way, as with .env, as from db with a special reding class. In both cases it could be personalized, I guess

So if it's more comfortable with .env - totally ok